### PR TITLE
Restructure repo and update readme

### PR DESCRIPTION
**Pull request by petedejoy**
This PR applies a high-level structure that more closely aligns with the UX of installing and accessing Airflow provider packages. this includes:
- Updating package naming and repo structure so that the UX of installing and importing the GE operator matches that of existing community operators. Users can now `pip install airflow-provider-great-expectations` and import operators via the `great_expectations_provider` module.
- Updating install and access instructions in readme.
- Placing the `examples` folder in the actual module folder so that they're built into the python wheel- this is how OSS providers have standardized (see https://github.com/apache/airflow/tree/master/airflow/providers/amazon/aws)

Still in MVP stages, but hope this helps make some progress, @spbail!

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -4,9 +4,9 @@
     long_description = fh.read()
 
 setuptools.setup(
-    name="great_expectations_airflow",
+    name="airflow-provider-great-expectations",
     version="0.0.1",
-    description="An Airflow operator for Great Expectations",
+    description="An Airflow provider for Great Expectations",
```
Comment (ashb) [inline setup.py:9]:
```suggestion
    description="An Airflow provider for Great Expectations",
    long_description=long_description,
    long_description_content_type='text/markdown',
```

Comment (petedejoy) [reply setup.py:9]: Good catch!

Comment (ashb) [inline setup.py:9]:
```suggestion
    description="An Apache Airflow provider for Great Expectations",
```

We should follow http://www.apache.org/foundation/marks/#guidelines

**Comment by ashb** [comment]
Oh, we possible need to also do something to include datafiles else the example CSV files won't be installed with the dist.

File: setup.py [hunk 2]
```diff
@@ -4,10 +4,13 @@
...
+    name="airflow-provider-great-expectations",
     version="0.0.1",
-    description="An Airflow operator for Great Expectations",
+    description="An Apache Airflow provider for Great Expectations",
+    long_description=long_description,
+    long_description_content_type='text/markdown',
     url="https://github.com/great-expectations/great_expectations_airflow",
     packages=setuptools.find_packages(),
     python_requires='>=3.6',
+    package_data={'': ['./great_expectations_provider/examples/data/*.csv']},
```
Comment (petedejoy) [inline setup.py:15]: @ashb is this what you're thinking re: including data files in the dist?

Comment (ashb) [reply setup.py]: Something like this, maybe yes.

If you run `python setup.py bdist_wheel && unzip -l dist/*.whl` you should see the .csv files in the right place. Do you?

Comment (petedejoy) [reply setup.py]: Fixed it up- I see 'em now :).

**Review by spbail — APPROVED** [review]
Thanks for making the changes! I'll merge them now, rename the repo, and test it out to see if there are any other tweaks needed...

**Pull request merged**